ui/admin/app/components/form/worker/create-worker-led/index.js

```diff
@@ -1,27 +1,44 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
 
 export default class FormWorkerCreateWorkerLedComponent extends Component {
-  // =attributes
+  // =services
+  @service features;
 
+  // =attributes
   @tracked generatedWorkerAuthToken;
   @tracked clusterId;
   @tracked ipAddress;
   @tracked configFilePath;
   @tracked workerTags;
+  @tracked initialUpstreams;
 
+  // =properties
+
+  /**
+   * Returns directory creation text for `<Rose::CodeEditor>`.
+   * @type {string}
+   */
   get createConfigText() {
     return `mkdir ${this.configFilePath || '<path>'}/ ;\\
 touch ${this.configFilePath || '<path>'}/pki-worker.hcl`;
   }
 
+  /**
+   * Returns config creation text for `<Rose::CodeEditor>`. The user will see
+   * different ouptuts based on if the are using `hcp` or `oss` binaries.
+   * @type {string}
+   */
   get workerConfigText() {
-    return `hcp_boundary_cluster_id = "${this.clusterId || '<config_id>'}"
-
-listener "tcp" {
+    return `${
+      this.features.isEnabled('byow-pki-hcp-cluster-id')
+        ? `hcp_boundary_cluster_id = "${this.clusterId || '<config_id>'}"`
+        : `listener "tcp" {
```

Comment:
Why is the listener part of the config being removed for the hcp edition?

Reply:
@zed yea that was mistake coming in the next commit

Reply:
@ZedLi inline conditional wasn't working for the `workerConfigText` getter, so I chose to do a simple `if and else` statement.  Adds a bit more code, but it allows control of spacing that is needed for the text output.